spec/wisper/activerecord/publisher_spec.rb

```diff
@@ -27,4 +31,70 @@ class User < ActiveRecord::Base
       user.destroy
     end
   end
+
+  context 'when specific broadcasts are specified' do
+    before do
+      User.broadcast_on :create
+    end
+
+    describe '#broadcast_create' do
+      it 'broadcasts event' do
+        user = User.new(name: 'Foo Bar')
+        expect(user).to receive(:broadcast).with('user_created', user)
+        user.save!
+      end
+    end
+
+    describe '#broadcast_update' do
+      it 'does not broadcast the event' do
+        user = User.create!(name: 'Foo Bar')
+        expect(user).not_to receive(:broadcast)
+        user.update!(name: 'Foo Bar Baz')
+      end
+    end
+
+    describe '#broadcast_destroy' do
+      it 'does not broadcast the event' do
+        user = User.create!(name: 'Foo Bar')
+        expect(user).not_to receive(:broadcast)
+        user.destroy
+      end
+    end
+  end
+
+  context 'when an unknown broadcast event is specified' do
+    it 'raises an ArgumentError' do
+      expect { User.broadcast_on :yolo }.to raise_error(ArgumentError)
+    end
+  end
+
+  context 'when no broadcasts are specified' do
+    before { User.disable_all_lifecycle_broadcasts! }
+
+    it 'never broadcasts events' do
+      user = User.new(name: 'Foo Bar')
+
+      expect(user).not_to receive(:broadcast)
+
+      user.save!
+      user.update!(name: 'Foo Bar Baz')
+      user.destroy
+    end
+  end
+
+  context 'when no broadcasts are specified' do
+    before { User.wisper_activerecord_publisher_broadcast_events = nil }
```

Comment:
is there a reason this is setting the events to `nil` vs using `disable_all_lifecycle_broadcasts!`?

Reply:
`disable_all_lifecycle_broadcasts!` sets `wisper_activerecord_publisher_broadcast_events = []` which is saying "this model has no broadcasts". setting `wisper_activerecord_publisher_broadcast_events = nil` is saying "this model has not defined any broadcasts" which means the default broadcasts are used instead.